db/resource.go

```diff
@@ -0,0 +1,33 @@
+package db
```

Comment:
I'm fine with it moving to another package, but I believe we should have this type to leverage the type system to say what this is, rather than doing hash based programming.